src/mlpack/methods/ann/layer/sigmoid_cross_entropy_error_impl.hpp

```diff
@@ -0,0 +1,68 @@
+/**
+ * @file sigmoid_cross_entropy_error_impl.hpp
+ * @author Kris Singh
+ *
+ * Implementation of the sigmoid cross entropy error performance function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SIGMOID_CROSS_ENTROPY_ERROR
+#define MLPACK_METHODS_ANN_LAYER_SIGMOID_CROSS_ENTROPY_ERROR
+
+// In case it hasn't yet been included.
+#include "sigmoid_cross_entropy_error.hpp"
+#include <mlpack/methods/ann/activation_functions/softplus_function.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+SigmoidCrossEntropyError<InputDataType, OutputDataType>
+::SigmoidCrossEntropyError()
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename eT>
+inline double SigmoidCrossEntropyError<InputDataType, OutputDataType>::Forward(
+    const arma::Mat<eT>&& input, const arma::Mat<eT>&& target)
+{
+  double maximum = 0;
+  for (size_t i = 0; i < input.n_elem; ++i)
+  {
+    maximum += std::max(input[i], 0.0) +
+        SoftplusFunction::Fn(-std::abs(input[i]));
```

Comment:
Since we deal with logit as input, we should directly use: ``std::log(1 + std::exp(x))`` here to avoid an unnecessary check.